### PR TITLE
Removing WATCH and WATCHLIST endpoints

### PR DESCRIPTION
**Pull request by yue9944882**
```release-note
Removing deprecated WATCH and WATCHLIST endpoints from apiserver
```

the deprecated endpoint has been hanging there for almost 10 minor releases, this pull chops them off completely as a follow-up of https://github.com/kubernetes/kubernetes/pull/41722

/kind cleanup
/sig api-machinery

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 1]
```diff
@@ -430,8 +420,6 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		actions = appendIf(actions, action{"LIST", resourcePath, resourceParams, namer, false}, isLister)
 		actions = appendIf(actions, action{"POST", resourcePath, resourceParams, namer, false}, isCreater)
 		actions = appendIf(actions, action{"DELETECOLLECTION", resourcePath, resourceParams, namer, false}, isCollectionDeleter)
-		// DEPRECATED in 1.11
-		actions = appendIf(actions, action{"WATCHLIST", "watch/" + resourcePath, resourceParams, namer, false}, allowWatchList)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go:434]: I would like to see this change, but I'm not sure if it's compatible with our policies...

@liggitt will be able to confirm

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/endpoints/installer.go:434]: it's in bounds per deprecation policies

it looks like some kubernetes org clients still use the deprecated watch paths - https://grep.app/search?q=watch/&filter[repo.pattern][0]=kubernetes-client/

**Pull request closed without merging**